static/js/util.ts

```diff
@@ -38,25 +43,28 @@ export function lower_bound(array, value, less) {
     return first;
 }
 
-export const lower_same = function lower_same(a, b) {
+export const lower_same = function lower_same(a: string, b: string): boolean {
     return a.toLowerCase() === b.toLowerCase();
 };
 
-export const same_stream_and_topic = function util_same_stream_and_topic(a, b) {
+export const same_stream_and_topic = function util_same_stream_and_topic(
+    a: Message,
+    b: Message,
+): boolean {
     // Streams and topics are case-insensitive.
     return a.stream_id === b.stream_id && lower_same(a.topic, b.topic);
 };
 
-export function is_pm_recipient(user_id, message) {
+export function is_pm_recipient(user_id: number, message: Message): boolean {
     const recipients = message.to_user_ids.split(",");
     return recipients.includes(user_id.toString());
 }
 
-export function extract_pm_recipients(recipients) {
+export function extract_pm_recipients(recipients: string): string[] {
     return recipients.split(/\s*[,;]\s*/).filter((recipient) => recipient.trim() !== "");
 }
 
-export const same_recipient = function util_same_recipient(a, b) {
+export const same_recipient = function util_same_recipient(a: Message, b: Message): boolean {
```

Comment:
Similarly, `same_recipient` is called with a non-`Message`:

https://github.com/zulip/zulip/blob/f5f6a3789bafcffb7441fddc669a83b5aca7c76f/static/js/compose_fade_helper.js#L8